src/exml.erl

```diff
@@ -53,6 +57,34 @@ xml_size({Key, Value}) ->
     + 4 % ="" and whitespace before
     + byte_size(Value).
 
+%% @doc Sort a (list of) `xmlel()`.
+%%
+%% Sorting is defined as calling `lists:sort/1` at:
+%% * all the `xmlel's provided (if there is a list of them) AND
+%% * all the `xmlel' elements' attributes recursively (the root and descendants) AND
+%% * all the `xmlel' children recursively (the root and descendants).
+%% The order is ascending.
+%%
+%% The implementation of this function is subtle modification of
+%% https://github.com/erszcz/rxml/commit/e8483408663f0bc2af7896e786c1cdea2e86e43d
+-spec xml_sort(item() | [item()]) -> item() | [item()].
+xml_sort(#xmlcdata{} = Cdata) ->
+    Cdata;
+xml_sort(#xmlel{} = El) ->
+    #xmlel{ attrs = Attrs, children = Children } = El,
+    El#xmlel{
+      attrs = lists:sort(Attrs),
+      children = lists:sort([ xml_sort(C) || C <- Children ])
```

Comment:
XML is order sensitive so sorting children can cause false positives in macro. I know that it *might* be useful to not care about the order sometimes but I'd prefer this to be less implicit, if we're going to keep it like this.

Reply:
Sorting children is the reason this macro exists. XML is not order sensitive - where did this idea come from? https://stackoverflow.com/questions/33746224/in-xml-is-the-attribute-order-important